### ISSUE TITLE
cli(diff-baseline): mi.* direction inverted in worsened/improved buckets

### ISSUE THREAD

**Issue by dekobon** [dekobon/big-code-analysis#825]
## Summary

`bca diff-baseline` buckets every metric increase as `worsened` and every
decrease as `improved`, ignoring metric direction — so for the lower-is-worse
`mi.*` (Maintainability Index) family the labels are inverted: an MI gain
(genuine improvement) is reported under **Worsened** and an MI drop (genuine
regression) under **Improved**.

## Location

- `big-code-analysis-cli/src/baseline_diff.rs:139-147` (`BaselineDiff::compute`)

## Evidence

`compute` classifies purely on the numeric direction of the value:

```rust
match n.value.partial_cmp(&o.value) {
    Some(Ordering::Greater) => worsened.push(value_delta(n, o.value, n.value)),
    Some(Ordering::Less) => improved.push(value_delta(n, o.value, n.value)),
    _ => {}
}
```

There is no consultation of metric direction. The rest of the CLI *does*
distinguish direction: `big_code_analysis::metric_catalog::lower_is_worse(id)`
(re-exported through `thresholds.rs::metric_is_lower_is_worse`) returns `true`
for the `mi.*` family, and the catalog marks `mi.original`, `mi.sei`,
`mi.visual_studio` as `Direction::LowerIsWorse`
(`src/metric_catalog.rs:150-152`). The threshold gate inverts its breach test
for exactly these metrics (`thresholds.rs:818`).

These three `mi.*` metrics are threshold extractors
(`big-code-analysis-cli/src/thresholds.rs:135-143`), so they are written into
`.bca-baseline.toml` and surface as `DiffEntry.metric` values that
`diff-baseline` pairs and buckets. The call site
(`commands.rs:2501`) passes `diff_entries()` straight into `compute` with no
direction adjustment.

The existing test `higher_value_is_worsened_lower_is_improved`
(`baseline_diff_tests.rs:46`) uses direction-agnostic metric names, so the
inversion for `mi.*` is unguarded.

## Expected Behavior

For a lower-is-worse metric (`mi.*`), a decrease in value should be reported as
**Worsened** and an increase as **Improved**, matching the semantics every
other consumer of these metrics already uses.

## Actual Behavior

For `mi.*` entries, an MI increase is bucketed `worsened` and an MI decrease is
bucketed `improved` — backwards. This also flips which rows survive
`--worsened-only` / `--improved-only`.

## Impact

Anyone running `bca diff-baseline` on baselines that contain `mi.*` metrics
(the default threshold set includes them) sees Maintainability Index changes
labelled with the opposite sentiment, and `--worsened-only`/`--improved-only`
filters select the wrong set. The summary counts ("N worsened, M improved")
are also wrong for these rows.

## Resolution (commit `d05a2f78`)

`BaselineDiff::compute` now resolves metric direction per entry via
`big_code_analysis::metric_catalog::lower_is_worse(&n.metric)` and
inverts the `Greater`/`Less` -> `worsened`/`improved` mapping for the
lower-is-worse `mi.*` family. `Equal`/NaN still contribute to no bucket.
Module + `ValueDelta` docs updated to note the inversion.

Regression coverage: `mi_family_direction_is_inverted` and
`mixed_directions_coexist_in_one_diff` (baseline_diff_tests.rs), plus the
end-to-end `mi_family_direction_is_inverted_end_to_end` (diff_baseline.rs,
#901). All verified to fail against the pre-fix sign-only logic via
test-via-revert. Left open for your review.

**Comment by dekobon** [comment]
## General Plan for Fix

### Root Cause

`BaselineDiff::compute` decides `worsened` vs `improved` solely from the sign
of `new.value - old.value`, encoding the higher-is-worse assumption that holds
for cyclomatic/cognitive/halstead/n*/loc but is exactly backwards for the
lower-is-worse `mi.*` family. The direction is already modelled centrally in
`big_code_analysis::metric_catalog` (`Direction::LowerIsWorse`,
`lower_is_worse(id)`), but the diff path never consults it.

### Implementation Steps

1. In `compute`, resolve direction per entry via
   `big_code_analysis::metric_catalog::lower_is_worse(&n.metric)` (or reuse the
   CLI re-export `thresholds::metric_is_lower_is_worse`) and flip the
   `Greater`/`Less` -> `worsened`/`improved` mapping when it returns `true`.
   Keep `Equal`/NaN contributing nothing.
2. Keep the change local to the bucket selection; `value_delta` already records
   old/new faithfully, so rows still display the true numbers.
3. Update the `//!` module doc (lines 8-11) which currently states
   "`worsened` = value rose, `improved` = value fell" unconditionally — note
   the lower-is-worse inversion.

### Tests to Add/Update

- Add a test in `baseline_diff_tests.rs` feeding an `mi.original` (or
  `mi.sei`/`mi.visual_studio`) entry that *increases* and asserting it lands in
  `improved`, and one that *decreases* asserting `worsened`.
- Add a mixed test (one `cyclomatic` rising + one `mi.original` rising) to
  confirm the two directions coexist in one diff.
- Verify-by-revert: confirm the new mi test fails against the current
  sign-only logic.

### Verification

- [ ] `cargo test -p big-code-analysis-cli` passes
- [ ] `cargo clippy -p big-code-analysis-cli` clean
- [ ] Manual: build two baselines differing only in an `mi.original` value
      (one up, one down) and confirm `bca diff-baseline` labels the gain as
      Improved and the drop as Worsened, and that `--improved-only` selects the
      gain.

**Comment by dekobon** [comment]
## Resolution (commit `d05a2f78`)

`BaselineDiff::compute` now resolves metric direction per entry via
`big_code_analysis::metric_catalog::lower_is_worse(&n.metric)` and
inverts the `Greater`/`Less` -> `worsened`/`improved` mapping for the
lower-is-worse `mi.*` family. `Equal`/NaN still contribute to no bucket.
Module + `ValueDelta` docs updated to note the inversion.

Regression coverage: `mi_family_direction_is_inverted` and
`mixed_directions_coexist_in_one_diff` (baseline_diff_tests.rs), plus the
end-to-end `mi_family_direction_is_inverted_end_to_end` (diff_baseline.rs,
#901). All verified to fail against the pre-fix sign-only logic via
test-via-revert. Left open for your review.
